packages/contracts-bedrock/scripts/Artifacts.s.sol

```diff
@@ -216,21 +216,26 @@ abstract contract Artifacts {
 
     /// @notice Returns the value of the internal `_initialized` storage slot for a given contract.
     function loadInitializedSlot(string memory _contractName) public returns (uint8 initialized_) {
-        address contractAddress;
-        // Check if the contract name ends with `Proxy` and, if so, get the implementation address
+        address contractAddress = mustGetAddress(_contractName);
+        string memory implementationName = _contractName;
+        StorageSlot memory slot;
+
+        // Check if the contract name ends with `Proxy` and, if so, override the contract address
+        // and implementation name.
         if (LibString.endsWith(_contractName, "Proxy")) {
+            implementationName = LibString.slice(_contractName, 0, bytes(_contractName).length - 5);
             contractAddress = EIP1967Helper.getImplementation(getAddress(_contractName));
```

Comment:
I thought that line 227 was the bug? `vm.load` is passed `contractAddress` which is always the implementation address

Reply:
yes, you're right. I restored the bug while trying to refactor for readability. 

But I've now realized that we don't need most of what is in here. The new version of this PR now simply overrides the contract name used to retrieve the storage layout.

1d1edb9e9782447e2d0d7bd939a9445282f63e0c